sw/device/silicon_creator/lib/shutdown.h

```diff
@@ -22,12 +23,14 @@ extern "C" {
  *
  * @param expr_ An expression which results in an rom_error_t.
  */
-#define SHUTDOWN_IF_ERROR(expr_) \
-  do {                           \
-    rom_error_t error = (expr_); \
-    if (error != kErrorOk) {     \
-      shutdown_finalize(error);  \
-    }                            \
+#define SHUTDOWN_IF_ERROR(expr_)           \
+  do {                                     \
+    rom_error_t error_ = expr_;            \
+    if (launder32(error_) != kErrorOk) {   \
+      HARDENED_CHECK_NE(error_, kErrorOk); \
```

Comment:
This isn't really necessary. We're already going to crash the chip, so the second check does not change program behavior under the presence of fault attacks.

Reply:
Yeah, I was on the fence about this one: shutdown_finalize will print `error_` if this check passes and print a different one if it fails. I thought I removed this but it seems like I didn't push the change.